### ISSUE TITLE
Generated test cases does not contaion example vaules

### ISSUE THREAD

**Issue by rkariyawas** [Cornutum/tcases#159]
Created test class using yaml file which contains the example values.
But generated test cases didn't contain the example values. 
Used command: mvn tcases:api-test
Used Tcases: 3.5.0 version

yaml file snippet:

`openapi: 3.0.1
info:
  title: User API
  description: This document outlines the user API contracts.
  contact:
    email: DL-test@test.com
  version: 2.0.0
externalDocs:
  url: 'https://test.test.com/display/rwar/User+API+Version+2'
tags:
  - name: Affiliation Operations
    description: Affiliation operations
  - name: Security Questions
    description: Fetch user security questions
  - name: User Operations
    description: All user operations
paths:
  '/v2/users/{userId}':
    get:
      tags:
        - User Operations
      summary: Finds user by user id or email
      operationId: findUser
      parameters:
        - name: userId
          in: path
          description: ID of user that needs to be fetched
          required: true
          schema:
            type: string
          examples:
            oneId:
              summary: example id
              value: [XZB-dsfdscfaWs-sdfsdfsv-dscsds-qw3ewdsc-asdsa]
        - name: x-request-id
          in: header
          description: Use a unique UID for this
          required: true
          schema:
            type: string
          example: traceId1698909
        - name: x-application-id
          in: header
          description: application id of the consuming application
          required: true
          schema:
            type: string
          example: alm-456
        - name: x-api-key
          in: header
          description: API key corresponding to the consuming application
          required: true
          schema:
            type: string`

**Comment by kerrykimbrough** [comment]
Here are some ways you can make it easier to investigate such issues.

1. Attach the complete OpenAPI spec file. You can do this in a comment by dragging and dropping (or cutting and pasting) the file into the comment box. You may have to add a *.txt extension so that it is accepted as a plain text file.
2. Check the messages in the log file (in this case, `tcases-api-test.log`) and report what you see there. Better yet, attach the log file. If you prefer, you can use the `-l stdout` option to print log messages to standard output.
3. Report the results that you think are incorrect. For example, list the output that contains the unexpected info.

**Comment by kerrykimbrough** [comment]
I see you are running with mvn, in which case log messages appear in the mvn command output.

**Comment by kerrykimbrough** [comment]
Note that to generate with examples, you need to run `mvn tcases:api-test -Dsource=examples`

**Comment by rkariyawas** [comment]
Hi Kerry,
Please note that I don't have examples for each attribute and its only for few attributes.
 And here I attached yaml file in text format and console log.

[Tcase-consolelog.txt](https://github.com/Cornutum/tcases/files/5651169/Tcase-consolelog.txt)

[sampleyaml.txt](https://github.com/Cornutum/tcases/files/5651170/sampleyaml.txt)

**Comment by rkariyawas** [comment]
Hi Kerry,

Also need to know why the test cases not generated for response fields in yaml file.
Test cases only generated for the requests.

**Comment by kerrykimbrough** [comment]
Yes, as the message below explains, no tests could be created for this GET request, because no examples were provided for the `x-api-key` parameter. Without a value for this parameter, no request can be executed. If you add at least one example for this parameter, then you'll see tests created.

```
[WARNING] User API,/v2/users/{userId},GET: Failed to create examples for x-api-key -- No example defined for schema of type=string. Ignoring this operation.
```

**Comment by kerrykimbrough** [comment]
About response tests: it depends on what you think a response test should be.

Does this mean a test that executes requests and makes assertions about the resulting responses? But such assertions are impossible for Tcases to generate. There is no way for Tcases to predict what the expected responses should be.

Does this mean a test that looks at all the different kinds of responses that could be returned by a request and verifies that the API client correctly handles all responses? But this kind of "response client test" is also impossible for Tcases to generate. For any particular response, there is no way for Tcases to know how to create a request that returns that response.

**Comment by kerrykimbrough** [comment]
Are you interested in doing "response client tests", as described above? If so, it might be possible for Tcases to create one form of this by "mocking" the API server. In other words, by not actually sending a request to a server but instead creating a mock response with specific contents.

**Comment by rkariyawas** [comment]
Regarding examples: In my spec has defined values only for selected attributes (Server defined values, arrays) and not for all attributes. So like to know is there way to generate test cases based on example availability.

**Comment by kerrykimbrough** [comment]
If I understand correctly, you are suggesting that when no example data is specified, Tcases should generate random input data (in the usual way). Yes, I think that may be possible. And it's a good idea. I will create a new issue for this.

**Comment by kerrykimbrough** [comment]
See issue #161

**Comment by rkariyawas** [comment]
Sure then it will be great.

**Comment by kerrykimbrough** [comment]
Resolved in release 3.5.1
